src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.csproj

```diff
@@ -63,175 +62,19 @@
-        /internalize ^
-        /xmldocs ^
-        /log:$(ILMergeResultDir)IlMergeLogResources.txt</IlmergeCommand>
-      <IlmergeCommand Condition="'$(SkipSigning)' != 'true' And Exists($(AssemblyOriginatorKeyFile))">$(IlmergeCommand) ^
-        /delaysign ^
-        /keyfile:$(AssemblyOriginatorKeyFile)</IlmergeCommand>
-      <IlmergeCommand>$(IlmergeCommand) ^
-        $(PathToBuiltNuGetPackResource) ^
-        @(ResourceArtifacts, ' ')</IlmergeCommand>
-    </PropertyGroup>
-    <Exec Command="$(IlmergeCommand)" ContinueOnError="false" />
-  </Target>
-
-  <!--These targets help get all the DLLs that are packed in the nuspec to be sent for signing.-->
-  <Target Name="GetIlmergeBuildOutput" DependsOnTargets="ILMergeNuGetPackResourcesPerCulture;_GetDllsInOutputDirectory" Returns="@(DllsToSign)">
-    <ItemGroup>
-      <DllsToSign Include="@(DllsInOutputDir)" KeepDuplicates="false" />
-    </ItemGroup>
-  </Target>
-
-  <Target Name="_GetDllsInOutputDirectory" Returns="@(DllsInOutputDir)">
-    <ItemGroup>
-      <DllsInOutputDir Include="$(OutputPath)ilmerge\*.dll" KeepDuplicates="false">
-        <StrongName>MsSharedLib72</StrongName>
-        <Authenticode>Microsoft400</Authenticode>
-      </DllsInOutputDir>
-      <DllsInOutputDir Include="$(OutputPath)ilmerge\*.xml" KeepDuplicates="false">
-        <Authenticode>MicrosoftXmlSHA2</Authenticode>
-      </DllsInOutputDir>
-    </ItemGroup>
-  </Target>
-
-  <!--These targets help get all the DLLs that are packed in the nuspec to be sent for signing.-->
-  <Target Name="GetIlmergeSymbolOutput" Returns="@(SymbolsToIndex)">
-    <ItemGroup>
-      <SymbolsToIndex Include="$(OutputPath)ilmerge\*.pdb" KeepDuplicates="false" />
-    </ItemGroup>
-  </Target>
-
-  <Target Name="GetFinalBuildOutput" Returns="@(DllsToIndex)">
-    <ItemGroup>
-      <DllsToIndex Include="$(OutputPath)ilmerge\NuGet.Build.Tasks.Pack.dll"/>
-    </ItemGroup>
-  </Target>
-
   <Target Name="CreatePackNupkg">
     <PropertyGroup>
-      <!-- Build from source can't use ILMerge and XPLat builds don't run the ILmerge targets. -->
-      <ILMergeSubpath Condition="'$(DotNetBuildSourceOnly)' != 'true' and '$(IsXPlat)' != 'true'">ilmerge\</ILMergeSubpath>
       <PackagePathDir Condition="'$(TargetFramework)' == '$(NETFXTargetFramework)'">Desktop/</PackagePathDir>
       <PackagePathDir Condition="'$(TargetFramework)' == '$(NetStandardVersion)'">CoreCLR/</PackagePathDir>
     </PropertyGroup>
     <ItemGroup>
-      <TfmSpecificPackageFile Include="$(OutputPath)\$(ILMergeSubpath)NuGet.Build.Tasks.Pack.dll" Condition="'$(DotNetBuildSourceOnly)' != 'true'">
+      <TfmSpecificPackageFile Include="$(OutputPath)\**\NuGet*.resources.dll">
         <PackagePath>$(PackagePathDir)</PackagePath>
       </TfmSpecificPackageFile>
-      <TfmSpecificPackageFile Include="$(OutputPath)\$(ILMergeSubpath)**\NuGet*.resources.dll" Exclude="$(OutputPath)\$(ILMergeSubpath)publish\**\NuGet*.resources.dll">
```

Comment:
Have you verified the content of the nupkg? IIRC it kept picking up more than what I expected.

Reply:
Yes, I updated the PR body with the structure.